eventdisplay/datasource/FairTimebasedSource.h

```diff
@@ -0,0 +1,42 @@
+/**
+ * FairTimebasedSource.h
+ *
+ * \brief Handles timebased data via the GetData method. A time window has to be given in which the data is received.
+ *
+ * \date 07.12.2020
+ * \author Tobias Stockmanns <t.stockmanns@fz-juelich.de>
+ */
+
+#ifndef EVENTDISPLAY_FAIRTIMEBASEDSOURCE_H_
+#define EVENTDISPLAY_FAIRTIMEBASEDSOURCE_H_
+
+#include "FairDataSourceI.h"
+#include "FairTSBufferFunctional.h"   // IWYU pragma: keep needed for cint
+#include "FairTimeStamp.h"
+
+class FairTimebasedSource : public FairDataSourceI
+{
+  public:
+    FairTimebasedSource();
+    FairTimebasedSource(TString branchName, Double_t windowMinus, Double_t windowPlus)
+        : FairDataSourceI(branchName)
+        , fTimeWindowMinus(windowMinus)
+        , fTimeWindowPlus(windowPlus){};
+    virtual ~FairTimebasedSource();
+
+    virtual void RetrieveData(double time);
+    virtual int GetNData() { return fCArray->GetEntries(); };
+    virtual TObject* GetData(int index) { return fCArray->At(index); }
+    virtual double GetTime(int index) { return static_cast<FairTimeStamp*>(fCArray->At(index))->GetTimeStamp(); }
+    virtual void Reset(){};
+
+  private:
+    BinaryFunctor* fStartFunctor{new StopTime()};   //!
```

Comment:
Who is deleting these objects? Having a brief look at FairRootManager and FairTSBufferFunctional, I don't see how these are deleted. @MohammadAlTurany @fuhlig1

Reply:
I guess you are speaking about the deletion of the `FairTSBufferFunctional` and not of the fStart/StopFunctors. Looking at the code I agree with you, that I have not found a delete for the FairTSBufferFunctionals nor for the containers used inside.
I assume that the root garbage collection takes over this task but it should be done in a more propper way.

Reply:
now i see the delete calls at eventdisplay/datasource/FairTimebasedSource.cxx:20, i guess i overlooked them before.

Reply:
But then the ownership is completely in the hands of this class and they can be made non-pointer members, and passed to ptr-requiring functions via `&`.